### PR TITLE
Reader: Add SyncReaderFollows component and use it

### PR DESCRIPTION
**Pull request by blowery**
Also adds in a lastSyncTime reducer, which the query component uses to determine when to resync the list

### CODE REVIEW

**Comment by blowery** [comment]
Pulled out of https://github.com/Automattic/wp-calypso/pull/13239

**Comment by samouri** [comment]
code LGTM. giving this a test-run now

File: client/components/data/sync-reader-follows/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Internal Dependencies
+ */
+import { getReaderFollowsLastSyncTime } from 'state/selectors';
+import { requestFollows } from 'state/reader/follows/actions';
+
+const TIME_BETWEEN_SYNCS = 1000 * 60 * 60; // one hour
+
+class SyncReaderFollows extends Component {
+	check() {
```
Comment (samouri) [inline client/components/data/sync-reader-follows/index.js]: how could we write a test for this?

Comment (blowery) [inline client/components/data/sync-reader-follows/index.js]: we could move the logic up to a selector, sure. that would make it easier to test.

Comment (blowery) [inline client/components/data/sync-reader-follows/index.js]: done

File: client/state/selectors/should-sync-reader-follows.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Internal Dependencies
+ */
+import { getReaderFollowsLastSyncTime } from 'state/selectors';
+
+const MS_BETWEEN_SYNCS = 1000 * 60 * 60; // one hour
+
+export default function shouldSyncReaderFollows( state ) {
+	const lastTime = getReaderFollowsLastSyncTime( state );
```
Comment (samouri) [inline client/state/selectors/should-sync-reader-follows.js]: whats the general best-practice for selectors using selectors?
it feels okay to me

File: client/state/selectors/test/should-sync-reader-follows.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * External Dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal Dependencies
+ */
+import { shouldSyncReaderFollows } from 'state/selectors';
+
+const TIME_BETWEEN_SYNCS = 1000 * 60 * 60; // one day in millis
```
Comment (samouri) [inline client/state/selectors/test/should-sync-reader-follows.js]: maybe make it an exported const from....the selector?

File: client/state/selectors/test/should-sync-reader-follows.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		expect( shouldSyncReaderFollows( {
+			reader: {
+				follows: {
+					lastSyncTime: null
+				}
+			}
+		} ) ).to.be.true;
+	} );
+
+	it( 'should return true when last time is just over an hour ago', () => {
```
Comment (samouri) [inline client/state/selectors/test/should-sync-reader-follows.js]: an hour or a day?

File: client/state/selectors/should-sync-reader-follows.js (hunk 2 above)
Comment (blowery) [inline client/state/selectors/should-sync-reader-follows.js]: should be fine as long as you don't make a dependency loop

File: client/state/selectors/test/should-sync-reader-follows.js (hunk 4 above)
Comment (blowery) [inline client/state/selectors/test/should-sync-reader-follows.js]: the constant is a an hour...

File: client/state/selectors/should-sync-reader-follows.js (hunk 2 above)
Comment (samouri) [inline client/state/selectors/should-sync-reader-follows.js]: ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿ 🔁 ➿

File: client/state/selectors/test/should-sync-reader-follows.js (hunk 3 above)
Comment (blowery) [inline client/state/selectors/test/should-sync-reader-follows.js]: done

File: client/state/selectors/test/should-sync-reader-follows.js (hunk 4 above)
Comment (samouri) [inline client/state/selectors/test/should-sync-reader-follows.js]: "const TIME_BETWEEN_SYNCS = 1000 * 60 * 60; // one day in millis"

Comment (samouri) [inline client/state/selectors/test/should-sync-reader-follows.js]: theres a lie in there somewhere!

Comment (blowery) [inline client/state/selectors/test/should-sync-reader-follows.js]: hah, removed anyway. exported the const in the selector.

**Pull request merged**